src/instructlab/utils.py

```diff
@@ -44,6 +44,37 @@ class TaxonomyReadingException(Exception):
     """An exception raised during reading of the taxonomy."""
 
 
+class Message(TypedDict):
+    """
+    Represents a message within an AI conversation.
+    """
+
+    content: str
+    role: str  # one of: 'user', 'assistant', or 'system'
```

Comment:
How about using an `Enum` for this? So that we can easily validate the presence of 'user', 'assistant', or 'system'. This could look like this:

```python
from typing import TypedDict, Optional
from enum import Enum

class Role(Enum):
    USER = 'user'
    ASSISTANT = 'assistant'
    SYSTEM = 'system'

class Message(TypedDict):
    """
    Represents a message within an AI conversation.
    """
    content: str
    role: Role

    def __post_init__(self):
        if self.role not in Role:
            raise ValueError(f"Invalid role: {self.role}")
```

Reply:
@leseb We shouldn't do this because `TypedDict` is intended for type-hinting and is just a dict under the hood. Therefore we wouldn't want the overhead of explicitly instantiating new objects for this type just to process a list.

Reply:
So I added it but I think it will be best to remove it. It's causing too many problems with MyPy and JSON serialization during unit testing. It'll be easier to just have it as a string with the comment above indicating its usage.

Reply:
I'm keeping it for now but look at the MyPy failure: https://github.com/instructlab/instructlab/actions/runs/9900406125/job/27351128053?pr=1644#step:8:30

The alternative is to use the key directly instead of using `.value`, but this causes a wider issue where JSON cannot serialize an enum directly. While we can technically add code to make this happen, it feels like we're adding a lot of complexity where we could instead just keep it simple.